ui/app/src/views/SwapPage.vue

```diff
@@ -110,6 +110,8 @@ export default defineComponent({
     }
 
     function swapInputs() {
+      // we want the "from" value to persist, but now in the "to" input
+      selectedField.value = "to";
```

Comment:
Should we also toggle the selected value, rather than just setting it to "to"?

Reply:
@59023g Also, I think we are generally not doing it correctly. This should swap both amounts and tokens (check Uniswap for reference). Otherwise, it does not make sense to swap these fields

Reply:
It does swap both amounts and tokens. Although the "from" amount updates. Which it was not before. This is the same behavior as Uniswap.

I will look into code suggestion